run_tests.sh

```diff
@@ -1,9 +1,13 @@
 #!/bin/bash -ex
 
 export PIPENV_IGNORE_VIRTUALENVS=1
+set +x  # Keep the API key out of execution logs
+export DEPSDEV_API_KEY="$(gcloud --project oss-vdb secrets versions access latest --secret depsdotdev-key)"
+test -z "$DEPSDEV_API_KEY" && echo "No API key for deps.dev" && exit 1
+set -x
 python3 -m pipenv sync
 python3 -m pipenv run python -m unittest osv.bug_test
-python3 -m pipenv run python -m unittest osv.ecosystems_test
+python3 -m pipenv run python -m unittest -v osv.ecosystems_test
```

Comment:
Is this -v part still needed? 

IF it's useful, is this something we want to apply to all?

Reply:
It's (in my opinion) useful in this instance so we know which tests are being skipped, given we have two sets of tests with two different criteria for being skipped. Happy to put it on all of them for consistency, I don't mind the verbose output myself...

```
+ python3 -m pipenv run python -m unittest -v osv.ecosystems_test
test_maven_deps_dev (osv.ecosystems_test.EnumerateTest.test_maven_deps_dev)
Test Maven using deps.dev. ... ok
test_alpine (osv.ecosystems_test.GetNextVersionTest.test_alpine)
Cloning into '/workspace/osv/testdata/version_enum/aports'...
Updating files:  89% (16158/18117)
Test Alpine ecosystem enumeration and caching behaviour ... ok
test_cache (osv.ecosystems_test.GetNextVersionTest.test_cache) ... ok
test_debian (osv.ecosystems_test.GetNextVersionTest.test_debian)
Test Debian ecosystem enumeration and caching behaviour ... WARNING:root:Package nginx has invalid fixed version: <end-of-life>. In debian release 9
ok
test_gems (osv.ecosystems_test.GetNextVersionTest.test_gems)
Test RubyGems. ... ok
test_maven (osv.ecosystems_test.GetNextVersionTest.test_maven)
Test Maven. ... skipped 'Unnecessary if using deps.dev'
test_maven_deps_dev (osv.ecosystems_test.GetNextVersionTest.test_maven_deps_dev)
Test Maven using deps.dev. ... ok
test_maven_with_cache (osv.ecosystems_test.GetNextVersionTest.test_maven_with_cache)
Test Maven. ... skipped 'Unnecessary if using deps.dev'
test_nuget (osv.ecosystems_test.GetNextVersionTest.test_nuget)
Test NuGet. ... ok
test_packagist (osv.ecosystems_test.GetNextVersionTest.test_packagist)
Test Packagist. ... ok
test_pypi (osv.ecosystems_test.GetNextVersionTest.test_pypi)
Test PyPI. ... ok
test_semver (osv.ecosystems_test.GetNextVersionTest.test_semver)
Test SemVer. ... ok

----------------------------------------------------------------------
Ran 12 tests in 80.321s

OK (skipped=2)
```

I'm satisfied the right tests are being skipped for the right reasons on Cloud Build.

Now, to your [point](https://github.com/google/osv.dev/pull/982#discussion_r1070848873) about external contributors, there's not a lot of value in them running a different set of tests, so should I just remove the non-deps.dev ones and have the inability to retrieve the secrets fail silently, meaning external contributors won't be able to test this?

Reply:
yeah, let's just have it fail (with a harmless log message) for external folks.

Reply:
It should do this in a40a406de8e0cd31f3c5eee473ddf603f341c2f1